control.py

```diff
@@ -17,69 +22,215 @@ def hello():
-        if valida_diretor_id(**diretor):
-            deletar_diretor(**diretor)
-            diretor_deletado = deletar_diretor(diretor("id"))
-            return jsonify(diretor_from_db_id(diretor_deletado))
-        else:
-            return jsonify({"Erro": "valor nao deletado"})
+            return jsonify({"Valor": "Nao adicionado a tabela"})
 
 
-@app.route("/diretores/<int:id>", methods=["PATCH"])
-def patch_diretor(id):
-    if request.method == "PATCH":
+@app.route("/diretores/<int:id>", methods=["PUT", "PATCH", "DELETE"])
+def diretor_delete_update(id):
+    if request.method == "DELETE":
+        try:
+            deletar_diretor(id)
+            return jsonify({"Valor": "Deletado"})
+        except:
+            return jsonify({"Valor": "Nao deletado"})
+    elif request.method == "PUT" or "PATCH":
         diretor = diretor_from_web(**request.json)
         if valida_diretor(**diretor):
             update_diretor(id, **diretor)
-            diretor_alterado = get_diretor(diretor(id))
+            diretor_alterado = get_diretor(id)
             return jsonify(diretor_from_db(diretor_alterado))
         else:
             return jsonify({"Erro": "Valor nao alterado"})
 
 
-@app.route("/usuarios", methods=["GET"])
-def usuarios_get():
-    nome_completo = nome_usuario_from_web(**request.args)
-    usuarios = select_usuarios(nome_completo)
-    usuarios_from_db = [usuario_from_db(usuario) for usuario in usuarios]
-    return jsonify(usuarios_from_db)
+@app.route("/usuarios", methods=["GET", "POST"])
+def usuario_getpost():
+    if request.method == "GET":
+        nome_completo = nome_usuario_from_web(**request.args)
+        usuarios = select_usuarios(nome_completo)
+        usuarios_from_db = [usuario_from_db(usuario) for usuario in usuarios]
+        return jsonify(usuarios_from_db)
+    elif request.method == "POST":
+        usuario = usuario_from_web(**request.json)
+        if valida_usuario(**usuario):
+            id_usuario = insert_usuario(**usuario)
+            usuario_cadastrado = get_usuario(id_usuario)
+            return jsonify(usuario_from_db(usuario_cadastrado))
+        else:
+            return jsonify({"erro": "Usuario invalido"})
+
+
+@app.route("/usuarios/<int:id>", methods=["PUT", "PATCH", "DELETE"])
+def usuario_delete_update(id):
+    if request.method == "DELETE":
+        try:
+            delete_usuario(id)
+            return jsonify({"Valor": "Deletado"})
+        except:
+            return jsonify({"Erro": "Valor nao alterado"})
+    elif request.method == "PUT" or "PATCH":
+        usuario = usuario_from_web(**request.json)
+        if valida_usuario(**usuario):
+            update_usuario(id, **usuario)
+            usuario_cadastrado = get_usuario(id)
+            return jsonify(usuario_from_db(usuario_cadastrado))
+        else:
+            return jsonify({"Erro": "Valor nao alterado"})
+
+
+@app.route("/generos", methods=["GET", "POST"])
+def genero_getpost():
```

Comment:
```suggestion
def genero_get_post():
```
Importante separar as palavras com _, e manter consistência na nomenclatura